integration_tests/src/main/python/hashing_test.py

```diff
@@ -174,4 +174,31 @@ def test_str_sha1():
 def test_str_special_characters_sha1():
     special_string_gen = StringGen().with_special_case('好').with_special_case('吃')
     assert_gpu_and_cpu_are_equal_collect(
-        lambda spark: unary_op_df(spark, special_string_gen).selectExpr('sha1(a)'))
\ No newline at end of file
+        lambda spark: unary_op_df(spark, special_string_gen).selectExpr('sha1(a)'))
+
+
+@pytest.mark.parametrize("datagen", [binary_gen,
+                                     string_gen,
+                                     StringGen().with_special_case('好').with_special_case('吃')])
```

Comment:
NIT: 
The third `StringGen` seems good enough, so the second string_gen is a little duplicate ?